### ISSUE TITLE
[THEORY] Issue with calculation of Qgen

### ISSUE THREAD

**Issue by victorventuri** [ROVI-org/thevenin#8]
Based on the [landing page](https://github.com/ROVI-org/thevenin) explanation, the calculation of Qgen is done by 
$$\dot{Q_{gen}} = I \times (V_{ocv}(soc) - V_{cell})$$

However, that assumes that the entire voltage drop between the OCV component and the terminals is due to heat dissipation power loss, no?. However, some of that energy "loss" is actually stored in the capacitors, that is, for each RC element, I believe the heat dissipation should be 
$$\dot{Q_{RC}} = I_R \times R_{RC}, $$
where $I_R$ is the current that flows through the resistor in the RC component. 

Is this reasoning correct?

**Comment by c-randall** [comment]
Equivalent circuit models (ECMs) attempt to capture the behavior of real physical systems without explicitly modeling any of their known processes. Rather than thinking about the circuit elements themselves, think about how the overpotentials they approximate relate to the true physics of the system: charge transfer reactions, ohmic/mass transport, etc.

The heat generation term can be derived starting from fundamental thermodynamics. It incorporates both reversible and irreversible processes [[1]](https://iopscience.iop.org/article/10.1149/1.1393625/meta) into an expression based on the total cell overpotential, i.e., $V_{\rm ocv}({\rm soc}) - V_{\rm cell}$. Other terms can show up as well, based on phase change, heat of mixing, and more. Here, we ignore some of the extra terms because they are either irrelevant to our system (there is no phase change) or they are small in comparison (e.g., heat of mixing).

If it helps to think about, this expression is also equivalent to 
```math
\dot{Q}_{\rm gen} = I^2 \times R_{\rm eff},
```
where $R_{\rm eff}$ is the total effective resistance for the circuit. In this form, there are no capacitors or other elements to think about. The effective resistance simply explains all overpotentials,
```math
V_{\rm ocv}({\rm soc}) - V_{\rm cell} = I \times R_{\rm eff}.
```

If you're interested, there's a short discussion about the expression we use [here](https://iopscience.iop.org/article/10.1149/2.042302jes), around equations 1 through 5.

**Comment by victorventuri** [comment]
I agree that energy balance dictates that difference between $(I\times\Delta V)$ and $\dot{Q}_{conv}$ has to be an internal $\dot{E}$. However, I think that not all of this $\dot{E}$ contributes to the heating of the system in the form of $m C_p \dot{T}$, as some of it is stored in capacitive-like elements (such as the formation of a double layer).

**Comment by c-randall** [comment]
Mathematically, I believe what you are suggesting is already equivalent to what I am using. I'm just using a form that is more common in the literature. We can see they are equivalent by considering one of the RC pairs. The overpotential is for RC pair $j$ is
```math
V_j = I_R \times R_j,
```
and that same overpotential evolves in time as
```math
C_j\frac{dV_j}{dt} = I_C.
```
From Kirchoff's law of currents, we also have $I = I_R + I_C$. These three expressions can be combined into a single ODE that describes how $V_j$ evolves with time,
```math
\frac{dV_j}{dt} = -\frac{V_j}{R_jC_j} + \frac{I}{C_j},
```
which matches what I am using for each RC overpotential. Consequently, each RC overpotential is equivalent to $I_R \times R_j$. This agrees with what I was saying before, each overpotential is related to a portion of the total "effective" resistance. In the case of an RC pair, the effective resistance ends up being $I_R \times R_j$. It follows that the Joule heating across this RC pair is the total current that moves across this overpotential, so $I \times V_j = I \times I_R \times R_j$. Since we already know the overpotential (it is solved for as a state variable) there is no need to calculate and explicitly use $I_R$ as you are suggesting.
